klayout/drc/rule_decks/main.drc

```diff
@@ -14,880 +14,766 @@
+#=======================================================================================================================
 require 'time'
-require "logger"
+require 'logger'
 require 'etc'
 
 exec_start_time = Time.now
 
-logger = Logger.new(STDOUT)
+logger = Logger.new($stdout)
 
-logger.formatter = proc do |severity, datetime, progname, msg|
-  "#{datetime}: Memory Usage (" + `pmap #{Process.pid} | tail -1`[10,40].strip + ") : #{msg}
+logger.formatter = proc do |_severity, datetime, _progname, msg|
+  "#{datetime}: Memory Usage (" + `pmap #{Process.pid} | tail -1`[10, 40].strip + ") : #{msg}
 "
 end
 
 #================================================
 #----------------- FILE SETUP -------------------
 #================================================
-logger.info("Starting running GF180MCU Klayout DRC runset on %s" % [$input])
-logger.info("Ruby Version for klayout: %s" % [RUBY_VERSION])
+logger.info(format('Starting running GF180MCU Klayout DRC runset on %s', $input))
+logger.info(format('Ruby Version for klayout: %s', RUBY_VERSION))
 
 if $input
-    if $topcell
-        source($input, $topcell)
-    else
-        source($input)
-    end
-end
-
-if $table_name
-  table_name = $table_name
-else
-  table_name = "main"
+  if $topcell
+    source($input, $topcell)
+  else
+    source($input)
+  end
 end
 
+table_name = $table_name || 'main'
 
-logger.info("Loading database to memory is complete.")
+logger.info('Loading database to memory is complete.')
 
 if $report
-    logger.info("GF180MCU Klayout DRC runset output at: %s" % [$report])
-    report("DRC Run Report at", $report)
+  logger.info(format('GF180MCU Klayout DRC runset output at: %s', $report))
+  report('DRC Run Report at', $report)
 else
-    layout_dir = Pathname.new(RBA::CellView::active.filename).parent.realpath
-    report_path = layout_dir.join("gf180_drc.lyrdb").to_s
-    logger.info("GF180MCU Klayout DRC runset output at default location: %s" % report_path)
-    report("DRC Run Report at", report_path)
+  layout_dir = Pathname.new(RBA::CellView.active.filename).parent.realpath
+  report_path = layout_dir.join('gf180_drc.lyrdb').to_s
+  logger.info('GF180MCU Klayout DRC runset output at default location: %s', report_path)
+  report('DRC Run Report at', report_path)
 end
 
 #================================================
 #------------------ SWITCHES --------------------
 #================================================
-logger.info("Evaluate switches.")
+logger.info('Evaluate switches.')
 
 # connectivity rules
-if $conn_drc == "true"
+if $conn_drc == 'true'
   CONNECTIVITY_RULES = $conn_drc
-  logger.info("connectivity rules are enabled.")
+  logger.info('connectivity rules are enabled.')
 else
   CONNECTIVITY_RULES = false
-  logger.info("connectivity rules are disabled.")
-end # connectivity rules
+  logger.info('connectivity rules are disabled.')
+end
 
 # WEDGE
-if $wedge == "false"
-  WEDGE = $wedge
-else
-  WEDGE = "true"
-end # WEDGE
+WEDGE = if $wedge == 'false'
```

Comment:
`$wedge ||= 'false'` ?

Reply:
Can we keep the original structure of the code then and keep any config refactoring for #37?

Reply:
Original structure has some linting issues.
![image](https://user-images.githubusercontent.com/91015308/217463997-5521b747-8d79-45e0-9590-924c7200fa6e.png)

Reply:
Ah I see, are you intentionally keeping this a `string` rather that converting it to a boolean? I would assume that the rest of the rules would prefer to test this directly rather than doing string comparison?

Reply:
Can you comment here ?

Reply:
Done, bool check function is used for string to boolean conversion.

![image](https://user-images.githubusercontent.com/91015308/218083493-758fb593-2223-4eff-98fa-51a5ef1a36be.png)

Reply:
![image](https://user-images.githubusercontent.com/91015308/218083596-dfa2e576-40dd-492b-9230-e4ab055950b8.png)